libsolidity/interface/FileReader.cpp

```diff
@@ -129,6 +178,41 @@ ReadCallback::Result FileReader::readFile(string const& _kind, string const& _so
 	}
 }
 
+string FileReader::cliPathToSourceUnitName(boost::filesystem::path const& _cliPath)
+{
+	vector<boost::filesystem::path> prefixes = {m_basePath.empty() ? normalizeCLIPathForVFS(".") : m_basePath};
+	prefixes += m_includePaths;
+
+	boost::filesystem::path normalizedPath = normalizeCLIPathForVFS(_cliPath);
+	for (boost::filesystem::path const& prefix: prefixes)
+		if (isPathPrefix(prefix, normalizedPath))
+		{
+			// Multiple prefixes can potentially match the path. We take the first one.
+			normalizedPath = stripPrefixIfPresent(prefix, normalizedPath);
+			break;
+		}
+
+	return normalizedPath.generic_string();
+}
+
+map<string, FileReader::FileSystemPathSet> FileReader::detectSourceUnitNameCollisions(FileSystemPathSet const& _cliPaths)
+{
+	map<string, FileReader::FileSystemPathSet> nameToPaths;
+	for (boost::filesystem::path const& cliPath: _cliPaths)
+	{
+		string sourceUnitName = cliPathToSourceUnitName(cliPath);
+		boost::filesystem::path normalizedPath = normalizeCLIPathForVFS(cliPath);
+		nameToPaths[sourceUnitName].insert(normalizedPath);
+	}
+
+	map<string, FileReader::FileSystemPathSet> collisions;
+	for (auto const& [sourceUnitName, cliPaths]: nameToPaths)
```

Comment:
```suggestion
	for (auto&& [sourceUnitName, cliPaths]: nameToPaths)
```
Wouldn't this have to be like this to move `cliPaths` below?

Reply:
Right. Thanks for spotting this! Fixed.